cmd2.py

```diff
@@ -2236,6 +2218,90 @@ def complete_shell(self, text, line, begidx, endidx):
         else:
             return path_complete(text, line, begidx, endidx)
 
+    def cmd_with_subs_completer(self, text, line, begidx, endidx, base):
```

Comment:
This is a very useful and convenient bit of code.  Thanks for the good work.

Now what I would love like to see in a follow-on PR is the following:
- unit tests covering this function
- update of [subcommands.py](https://github.com/python-cmd2/cmd2/blob/master/examples/subcommands.py) and/or [tab_completion.py](https://github.com/python-cmd2/cmd2/blob/master/examples/tab_completion.py) example to make use of this
- update of Sphinx documentation [sub-commands section](http://cmd2.readthedocs.io/en/latest/argument_processing.html?#sub-commands)